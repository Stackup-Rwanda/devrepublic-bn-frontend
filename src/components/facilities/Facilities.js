import React, { Component, createRef } from 'react';
import
{
  Container, Row, Col, Button,
} from 'react-bootstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import axios from 'axios';
import { toast } from 'react-toastify';
import { Link } from 'react-router-dom';
import NavBar from '../sharedComponents/AuthNavBar';
import Footer from '../sharedComponents/Footer';
import SideBar from '../sharedComponents/Sidebar';
import axiosOption from '../../util/axiosOption';
import translate from '../languages/Translate';
import Item from './FacilityItem';
import { getRequestsAction } from '../../redux/actions/getRequests';
import { getUserInfo } from '../../redux/actions/userActions';
import { getFacilities } from '../../redux/actions/facilityActions';
import '../../scss/facilities.scss';

class Facilities extends Component {
  constructor(props) {
    const today = new Date();
    const todayDate = today.toISOString().substr(0, 10);
    super(props);
    this.state = {
      show: false,
      facilityId: null,
      facilityName: null,
      rooms: [],
      roomId: '',
      checkin: todayDate,
      checkout: todayDate,
      location: null,
    };
    this.bookErrorRef = createRef();
  }

  async componentDidMount() {
    const token = localStorage.getItem('token');
    await this.props.dispatch(getUserInfo(token));
    await this.props.dispatch(getRequestsAction(token));
    await this.props.dispatch(getFacilities(token));
  }

setPopupValue = (data) => {
  this.setState({
    show: true, facilityId: data.id, rooms: data.rooms, facilityName: data.facilityName, location: data.location,
  });
}

handleChange = (event) => {
  const { value, name } = event.target;
  this.setState({ [name]: value });
}

resetBooking = () => {
  const today = new Date();
  const todayDate = today.toISOString().substr(0, 10);
  this.setState({
    facilityId: null, rooms: [], show: false, roomId: '', checkin: todayDate, checkout: todayDate, facilityName: null,
  });
}

handleBooking = async (event) => {
  const {
    roomId, facilityId, checkin, checkout, facilityName, location,
  } = this.state;
  event.preventDefault();
  if (!roomId) {
    this.bookErrorRef.current.style.display = 'block';
    return setTimeout(() => { this.bookErrorRef.current.style.display = 'none'; }, 2000);
  }
  const request = this.props.requests.find((el) => el.destination === location);
  if (!request) {
    return toast.error('please create a request to this destination before booking', {
      autoClose: 9000,
      position: 'top-right',
    });
  }
  try {
    await axios.post(`${process.env.BACKEND_LINK}/api/v1/facilities/book`, {
      facilityId, roomId, checkin, checkout, facilityName, requestId: request.id,
    }, axiosOption(localStorage.getItem('token')));
    await this.props.dispatch(getFacilities(localStorage.getItem('token')));
    this.resetBooking();
    return toast.info(`a booking to ${facilityName} has been created`, {
      autoClose: 9000,
      position: 'top-right',
    });
  } catch (error) {
    return toast.error(error.response.data.error, {
      autoClose: 9000,
      position: 'top-right',
    });
  }
}

likeUnlikeFacility = async (id, type) => {
  const token = localStorage.getItem('token');
  try {
    await axios.patch(`${process.env.BACKEND_LINK}/api/v1/facilities/${type}?id=${id}`, {}, axiosOption(token));
    await this.props.dispatch(getFacilities(token));
  } catch (error) {
    toast.error('unable to like or unlike the facility', {
      autoClose: 9000,
      position: 'top-right',
    });
  }
}

render() {
  const {
    show, rooms, roomId, checkin, checkout,
  } = this.state;
  const { intl, role, id } = this.props;
  const facilities = this.props.facilities.map((el) => (<Item key={el.id} facility={el} setValue={this.setPopupValue} role={role} userId={id} show={show} likeUnlikeFacility={this.likeUnlikeFacility} />));
  const roomAvailable = rooms.filter((el) => el.availability);
  const roomOption = roomAvailable.map((el) => (
    <option key={el.id} value={el.id}>
      {el.roomName}
      {' '}
      {el.type}
    </option>
  ));
  const today = new Date();
  const todayDate = today.toISOString().substr(0, 10);
  return (
    <>
      <SideBar>
        <div className="nav-item">
          <Link to="/profile">{translate('Profile')}</Link>
        </div>
        <div className="nav-item">
          <Link to="/dashboard">
            {translate('Travel Requests')}
          </Link>
        </div>
        <div className="nav-item same-page">
          <Link to="/facilities">
            {translate('Facilities')}
          </Link>
        </div>
      </SideBar>
      <NavBar />
      <Container className="facility-container">
        <Row>
          <Col className="facilities-title">{translate('Facilities available')}</Col>
        </Row>
        <Row className="facility-card-row">
          {facilities}
        </Row>
        <form style={{ display: show ? 'block' : 'none' }} className="book-popup shadow p-3 mb-5 bg-white rounded">
          <div className="book-error" ref={this.bookErrorRef}>{translate('please choose a room first')}</div>
          <div className="rooms">
            <span>{translate('available rooms')}</span>
            <select name="roomId" defaultValue={roomId} onChange={this.handleChange}>
              <option disabled value="">{intl.formatMessage({ id: 'available rooms', value: 'available rooms' })}</option>
              {roomOption}
            </select>
          </div>
          <div className="checkin">
            <span>{translate('Checkin')}</span>
            <input type="date" min={todayDate} value={checkin} name="checkin" onChange={this.handleChange} />
          </div>
          <div className="checkout">
            <span>{translate('Checkout')}</span>
            <input type="date" min={todayDate} value={checkout} name="checkout" onChange={this.handleChange} />
          </div>
          <div className="book-button">
            <Button
              className="cancel-booking"
              onClick={this.resetBooking}
            >
              {translate('Cancel')}
            </Button>
            <Button className="booking-button" onClick={this.handleBooking}>{translate('Book')}</Button>
          </div>
        </form>
      </Container>
      <Footer />
    </>

  );
}
}

Facilities.propTypes = {
  facilities: PropTypes.array,
  dispatch: PropTypes.func,
  requests: PropTypes.array,
  intl: PropTypes.object,
  role: PropTypes.string,
  id: PropTypes.string,
};

const mapStateToProps = ({ facilities, requests, profile }) => ({
  facilities: facilities.facilities,
  facilityServerError: facilities.error,
  requests: requests.requests,
  role: profile.role,
  id: profile.id,
});

export default injectIntl(connect(mapStateToProps)(Facilities));
