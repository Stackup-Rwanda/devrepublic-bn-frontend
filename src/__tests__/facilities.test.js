import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { IntlProvider } from 'react-intl';
import Facilities from '../components/facilities/Facilities';
import translation from '../components/languages/en.json';

jest.mock('socket.io-client', () => {
  const emit = jest.fn();
  const on = jest.fn(() => [{
    id: 'fafwefwefwefw', content: 'this is a comment', createdAt: '2019-02-02', status: 'read',
  }]);
  const socket = { emit, on };
  return jest.fn(() => socket);
});

const mockStore = configureStore([]);
const dispatch = jest.fn();
const store = mockStore({
  facilities: {
    facilities: [{
      id: '5be72db7-5510-4a50-9f15-e23f103116d5',
      facilityName: 'Marriot',
      location: 'Nairobi',
      image: '',
      createdBy: '0119b84a-99a4-41c0-8a0e-6e0b6c385165',
      numOfRooms: 4,
      amenities: null,
      services: null,
      likes: 0,
      totalRating: 0,
      averageRating: 0,
      likesId: [],
      unlikes: 0,
      unlikesId: [],
      createdAt: '2020-06-03T11:22:18.966Z',
      updatedAt: '2020-06-03T12:26:54.612Z',
      rooms: [
        {
          id: '4ef246b8-b38a-44b6-9072-90b1763c0a76',
          facilityId: '5be72db7-5510-4a50-9f15-e23f103116d5',
          roomName: 'high',
          type: 'queen size',
          availability: false,
          createdAt: '2020-06-03T12:26:54.606Z',
          updatedAt: '2020-06-03T12:30:47.798Z',
        },
      ],
    }],
    facilitiesServerError: null,
  },
  profile: {
    role: 'Requester',
    image: 'www.image.example',
  },
  requests: {
    requests: [],
  },
  user: {
    user: 'eeferferfer',
  },
  language: {
    language: 'en',
  },
  dispatch,
});
const store2 = mockStore({
  facilities: {
    facilities: [{
      id: '5be72db7-5510-4a50-9f15-e23f103116d5',
      facilityName: 'Marriot',
      location: 'Nairobi',
      image: '',
      createdBy: '0119b84a-99a4-41c0-8a0e-6e0b6c385165',
      numOfRooms: 4,
      amenities: null,
      services: null,
      likes: 2000,
      totalRating: 0,
      averageRating: 0,
      likesId: [],
      unlikes: 0,
      unlikesId: [],
      createdAt: '2020-06-03T11:22:18.966Z',
      updatedAt: '2020-06-03T12:26:54.612Z',
      rooms: [
        {
          id: '4ef246b8-b38a-44b6-9072-90b1763c0a76',
          facilityId: '5be72db7-5510-4a50-9f15-e23f103116d5',
          roomName: 'high',
          type: 'queen size',
          availability: false,
          createdAt: '2020-06-03T12:26:54.606Z',
          updatedAt: '2020-06-03T12:30:47.798Z',
        },
      ],
    }],
    facilitiesServerError: null,
  },
  profile: {
    role: 'Requester',
    image: 'www.image.example',
  },
  requests: {
    requests: [{
      destination: 'Nairobi',
      requestId: 'fvevefergegwerg',
    }],
  },
  user: {
    user: 'eeferferfer',
  },
  language: {
    language: 'en',
  },
  dispatch,
});

describe('<Facilities>', () => {
  it('render correctly Facilities component', () => {
    const container = (
      <IntlProvider defaultLocale="en" locale="en" messages={translation}>
        <MemoryRouter>
          <Provider store={store}>
            <Facilities intl={{ formatMessage: jest.fn() }} />
          </Provider>
        </MemoryRouter>

      </IntlProvider>
    );
    const DateInputComponent = renderer.create(container).toJSON();
    expect(DateInputComponent).toMatchSnapshot();
  });
  it('it should open a pop when use click on the book button', () => {
    const container = mount(
      <IntlProvider defaultLocale="en" locale="en" messages={translation}>
        <MemoryRouter>
          <Provider store={store}>
            <Facilities intl={{ formatMessage: jest.fn() }} />
          </Provider>
        </MemoryRouter>

      </IntlProvider>,
    );
    const element = container.childAt(0).childAt(0).childAt(0).childAt(0)
      .childAt(0)
      .children();
    const bookButton = element.find('.book-facility');
    bookButton.simulate('click');
    expect(element.instance().state.show).toBeTruthy();
    expect(element.instance().state.facilityId).toBe('5be72db7-5510-4a50-9f15-e23f103116d5');
    expect(element.instance().state.facilityName).toBe('Marriot');
    expect(element.instance().state.location).toBe('Nairobi');
    expect(element.instance().state.rooms[0].roomName).toBe('high');
    expect(element.instance().state.rooms[0].id).toBe('4ef246b8-b38a-44b6-9072-90b1763c0a76');
    expect(element.instance().state.roomId).toBe('');
    expect(element.instance().state.checkin).toBe((new Date()).toISOString().substr(0, 10));
    expect(element.instance().state.checkout).toBe((new Date()).toISOString().substr(0, 10));
  });
  it('it should change the state when the input value are change', () => {
    const container = mount(
      <IntlProvider defaultLocale="en" locale="en" messages={translation}>
        <MemoryRouter>
          <Provider store={store}>
            <Facilities intl={{ formatMessage: jest.fn() }} />
          </Provider>
        </MemoryRouter>

      </IntlProvider>,
    );
    const element = container.childAt(0).childAt(0).childAt(0).childAt(0)
      .childAt(0)
      .children();
    const bookButton = element.find('.book-facility');
    const checkIn = element.find('input[name="checkin"]');
    const checkOut = element.find('input[name="checkout"]');
    bookButton.simulate('click');
    checkIn.simulate('change', { target: { value: '2020-07-07', name: 'checkin' } });
    checkOut.simulate('change', { target: { value: '2020-07-17', name: 'checkout' } });
    expect(element.instance().state.checkin).toBe('2020-07-07');
    expect(element.instance().state.checkout).toBe('2020-07-17');
  });
  it('it should cancel book when user click on Cancel', () => {
    const container = mount(
      <IntlProvider defaultLocale="en" locale="en" messages={translation}>
        <MemoryRouter>
          <Provider store={store}>
            <Facilities intl={{ formatMessage: jest.fn() }} />
          </Provider>
        </MemoryRouter>

      </IntlProvider>,
    );
    const element = container.childAt(0).childAt(0).childAt(0).childAt(0)
      .childAt(0)
      .children();
    const bookButton = element.find('.book-facility');
    const cancelButton = element.find('.cancel-booking button');
    bookButton.simulate('click');
    cancelButton.simulate('click');
    expect(element.instance().state.show).toBeFalsy();
    expect(element.instance().state.rooms).toEqual([]);
    expect(element.instance().state.facilityId).toBe(null);
    expect(element.instance().state.facilityName).toEqual(null);
  });
  it('it should cancel book when user click on Cancel', () => {
    const container = mount(
      <IntlProvider defaultLocale="en" locale="en" messages={translation}>
        <MemoryRouter>
          <Provider store={store}>
            <Facilities intl={{ formatMessage: jest.fn() }} />
          </Provider>
        </MemoryRouter>

      </IntlProvider>,
    );
    const element = container.childAt(0).childAt(0).childAt(0).childAt(0)
      .childAt(0)
      .children();
    const bookButton = element.find('.book-facility');
    const bookingButton = element.find('.booking-button button');
    bookButton.simulate('click');
    bookingButton.simulate('click');
    expect(element.instance().state.show).toBeTruthy();
  });
  it('like a facility', () => {
    const container = mount(
      <IntlProvider defaultLocale="en" locale="en" messages={translation}>
        <MemoryRouter>
          <Provider store={store}>
            <Facilities intl={{ formatMessage: jest.fn() }} />
          </Provider>
        </MemoryRouter>

      </IntlProvider>,
    );
    const element = container.childAt(0).childAt(0).childAt(0).childAt(0)
      .childAt(0)
      .children();
    const likeButton = element.find('.like-button');
    likeButton.simulate('click');
    expect(dispatch.mock.calls.length).toBe(0);
    expect(element.instance().state.show).toBeFalsy();
    expect(element.find('.total-likes').instance().textContent).toBe('0');
  });
  it('it should not book when the check in date equals the checkout date', () => {
    const container = mount(
      <IntlProvider defaultLocale="en" locale="en" messages={translation}>
        <MemoryRouter>
          <Provider store={store}>
            <Facilities intl={{ formatMessage: jest.fn() }} />
          </Provider>
        </MemoryRouter>

      </IntlProvider>,
    );
    const element = container.childAt(0).childAt(0).childAt(0).childAt(0)
      .childAt(0)
      .children();
    const bookButton = element.find('.book-facility');
    const select = element.find('select');
    const bookingButton = element.find('.booking-button button');
    bookButton.simulate('click');
    select.simulate('change', { target: { value: '4ef246b8-b38a-44b6-9072-90b1763c0a76', name: 'roomId' } });
    bookingButton.simulate('click');
    expect(element.instance().state.roomId).toBe('4ef246b8-b38a-44b6-9072-90b1763c0a76');
  });
  it('it should not book when the check in date doesn\'t equals the checkout date but the user hasn\'t visited the facility', () => {
    const container = mount(
      <IntlProvider defaultLocale="en" locale="en" messages={translation}>
        <MemoryRouter>
          <Provider store={store2}>
            <Facilities intl={{ formatMessage: jest.fn() }} />
          </Provider>
        </MemoryRouter>

      </IntlProvider>,
    );
    const element = container.childAt(0).childAt(0).childAt(0).childAt(0)
      .childAt(0)
      .children();
    const bookButton = element.find('.book-facility');
    const select = element.find('select');
    const bookingButton = element.find('.booking-button button');
    bookButton.simulate('click');
    select.simulate('change', { target: { value: '4ef246b8-b38a-44b6-9072-90b1763c0a76', name: 'roomId' } });
    bookingButton.simulate('click');
    expect(element.instance().state.roomId).toBe('4ef246b8-b38a-44b6-9072-90b1763c0a76');
    expect(element.find('.total-likes').instance().textContent).toBe('2.00k');
  });
});
