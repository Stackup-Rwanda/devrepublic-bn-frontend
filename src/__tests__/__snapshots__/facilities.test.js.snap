// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Facilities> render correctly Facilities component 1`] = `
Array [
  <div
    className="sidebar-wrapper"
  >
    <div
      className="sidebar"
      style={
        Object {
          "display": "block",
        }
      }
    >
      <div
        className="nav-item"
      >
        <a
          href="/stats"
          onClick={[Function]}
        >
          Overview
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          Profile
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          href="/dashboard"
          onClick={[Function]}
        >
          Travel Requests
        </a>
      </div>
      <div
        className="nav-item same-page"
      >
        <a
          href="/facilities"
          onClick={[Function]}
        >
          Facilities
        </a>
      </div>
    </div>
    <button
      onClick={[Function]}
      style={
        Object {
          "left": "194px",
        }
      }
      type="button"
    >
      close
    </button>
  </div>,
  <nav
    className="nav-bar navbar navbar-expand-lg navbar-light fixed-top"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="barefoot nomad logo"
        className="d-inline-block align-top"
        height="30"
        src="test-file-stub"
        width="30"
      />
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="nav-bar_items navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      />
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-bar_btn text-center btn btn-primary"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
           
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <div
            className="notification-wrapper"
          >
            <button
              id="bell-button"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="bell"
                className="notification-bell"
                src="test-file-stub"
              />
            </button>
            <div
              className="notif-panel container"
              style={
                Object {
                  "display": "none ",
                  "height": "280px",
                }
              }
            >
              <p>
                Notifications
                <span
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  all notification already seen
                </span>
              </p>
              <div
                className="notifications"
                style={null}
              />
              <div
                className="row notif-buttons"
              >
                <button
                  className="col"
                  id="mark-all"
                  onClick={[Function]}
                  type="button"
                >
                  mark all as read
                </button>
                <button
                  className="col"
                  id="view-more"
                  onClick={[Function]}
                  type="button"
                >
                  view more
                </button>
              </div>
            </div>
          </div>
        </a>
        <img
          alt="profile"
          className="nav-profile-img"
          src="www.image.example"
        />
        <div
          className="select-lang dropdown"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-bar_dropdown nav-bar_dropdown-inherit"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            en
          </button>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="facility-container container"
  >
    <div
      className="row"
    >
      <div
        className="facilities-title col"
      >
        Facilities available
      </div>
    </div>
    <div
      className="facility-card-row row"
    >
      <div
        className="facility-card"
      >
        <div
          className="average-rating"
        >
          <svg
            enableBackground="new 0 0 24 24"
            height="25"
            id="color"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
              fill="#ffc107"
            />
          </svg>
          <svg
            enableBackground="new 0 0 24 24"
            height="25"
            id="color"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
              fill="#ffc107"
            />
          </svg>
          <svg
            enableBackground="new 0 0 24 24"
            height="25"
            id="color"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
              fill="#ffc107"
            />
          </svg>
          <svg
            enableBackground="new 0 0 24 24"
            height="25"
            id="color"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
              fill="#ffc107"
            />
          </svg>
          <svg
            enableBackground="new 0 0 24 24"
            height="25"
            id="color"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m23.363 8.584-7.378-1.127-3.307-7.044c-.247-.526-1.11-.526-1.357 0l-3.306 7.044-7.378 1.127c-.606.093-.848.83-.423 1.265l5.36 5.494-1.267 7.767c-.101.617.558 1.08 1.103.777l6.59-3.642 6.59 3.643c.54.3 1.205-.154 1.103-.777l-1.267-7.767 5.36-5.494c.425-.436.182-1.173-.423-1.266z"
              fill="#ffc107"
            />
          </svg>
        </div>
        <div
          className="facility-image"
        >
          <img
            alt="profile"
            className="facility-iamge"
            src="test-file-stub"
          />
        </div>
        <div
          className="facility-info"
        >
          <span
            className="facility-name"
          >
            Marriot
          </span>
          <span
            className="facility-location"
          >
            Nairobi
          </span>
        </div>
        <div
          className="rating-likes"
        >
          <div
            className="likes"
          >
            <button
              type="button"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  className="like-svg"
                  d="M4 21h1V8H4a2 2 0 0 0-2 2v9a2 2 0 0 0 2 2zM20 8h-7l1.122-3.368A2 2 0 0 0 12.225 2H12L7 7.438V21h11l3.912-8.596L22 12v-2a2 2 0 0 0-2-2z"
                  fill="#065471"
                />
              </svg>
            </button>
            <button
              type="button"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  className="like-svg"
                  d="M20 3h-1v13h1a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2zM4 16h7l-1.122 3.368A2 2 0 0 0 11.775 22H12l5-5.438V3H6l-3.937 8.649l-.063.293V14a2 2 0 0 0 2 2z"
                  fill="#065471"
                />
              </svg>
            </button>
          </div>
          <div
            className="ratings"
          >
            <button
              type="button"
            >
              Add rating
            </button>
          </div>
        </div>
        <button
          className="book-facility"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="button"
        >
          Book
        </button>
      </div>
    </div>
    <form
      className="book-popup shadow p-3 mb-5 bg-white rounded"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="book-error"
      >
        please choose a room first
      </div>
      <div
        className="rooms"
      >
        <span>
          available rooms
        </span>
        <select
          defaultValue=""
          name="roomId"
          onChange={[Function]}
        >
          <option
            disabled={true}
            value=""
          >
            available rooms
          </option>
        </select>
      </div>
      <div
        className="checkin"
      >
        <span>
          Checkin
        </span>
        <input
          min="2020-06-05"
          name="checkin"
          onChange={[Function]}
          type="date"
          value="2020-06-05"
        />
      </div>
      <div
        className="checkout"
      >
        <span>
          Checkout
        </span>
        <input
          min="2020-06-05"
          name="checkout"
          onChange={[Function]}
          type="date"
          value="2020-06-05"
        />
      </div>
      <div
        className="book-button"
      >
        <button
          className="cancel-booking btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="booking-button btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Book
        </button>
      </div>
    </form>
  </div>,
  <div
    className="fixed-bottom footer container-fluid"
  >
    <div
      className="justify-content-md-center row"
    >
      <div
        className="footer_text col-md-auto"
      >
        Copyright 2020 devrepublic team. All rights reserved.
      </div>
    </div>
  </div>,
]
`;
